### DISCUSSION TITLE
Vanna is sending multiple queries (including the presidencies sql)

### DISCUSSION THREAD

**Discussion by merhawi25** [vanna-ai/vanna#640] [Q&A] [answered]
Tools:
LLM: OPENAI gpt-4o
Vectordb: chromadb
Query DB: presto

**First question:** What is the full name of the customer who has paid the most?
**Vanna AI response:** ``` SELECT c.first_name, c.last_name
FROM raw.sales.customers c
JOIN raw.sales.orders o ON c.customer_id = o.customer_id
JOIN raw.sales.payments p ON o.order_id = p.order_id
GROUP BY c.customer_id, c.first_name, c.last_name
ORDER BY SUM(p.amount) DESC
LIMIT 1;```

**right/wrong:** right.

Me -> second question: who has paid the least?

**Vanna AI:**  "I interpreted your question as: What are the full names of the customers who have paid the most and the least?

 Edit Question New Question
-- Customer who has paid the most
select c.customer_id, c.first_name, c.last_name, sum(p.amount) as total_amount
from raw.sales.customers c
join raw.sales.orders o on c.customer_id = o.customer_id
join raw.sales.payments p on o.order_id = p.order_id
group by c.customer_id, c.first_name, c.last_name
order by total_amount desc
limit 1;

-- Customer who has paid the least
select c.customer_id, c.first_name, c.last_name, sum(p.amount) as total_amount
from raw.sales.customers c
join raw.sales.orders o on c.customer_id = o.customer_id
join raw.sales.payments p on o.order_id = p.order_id
group by c.customer_id, c.first_name, c.last_name
order by total_amount asc
limit 1;"

**right/wrong:** wrong

I have faced this crazy issue. I don't understand why it's sending the previous sql? or I'm missed something? Of course this is very serious issue.

**Comment by abhishektomar28** [comment]
I am also facing the same issue. I think every time need to start the new question.

**Comment by saurabhsbora** [comment]
yes I have also faced this issue where if you continue asking different questions in the same chat thread, it tries to combine all the questions into one question & generate answer accordingly. So every time we need to start a new question so that the question is answered independently. Don't know the logic behind this implementation

**Comment by henrywang0314** [comment, marked as answer]
class YourVannaClass(ChromaDB_VectorStore, OpenAI_Chat):
     ...

    def generate_rewritten_question(self, last_question: str, new_question: str, **kwargs) -> str:
        return new_question

this will override generate_rewritten_question under vannabase class
